### PR TITLE
Feature/integration specs

### PR DESCRIPTION
**Pull request by PNdunguMaina**
In this project, I did the following: ⚒️ 

- [ ] Wrote integration tests with Capybara gem.
- [ ] Secured app from n+1 problems with Bullet gem.

### CODE REVIEW

File: spec/views/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    it 'displays a button that lets me view all of a user posts' do
+      visit "/users/#{@first_user.id}/"
+      expect(page).to have_link('See all posts')
+    end
+
+    it 'When I click to see all posts, it redirects me to the user post index page' do
+      visit "/users/#{@first_user.id}/"
+      click_link 'See all posts'
+      expect(page).to have_current_path("/users/#{@first_user.id}/posts/")
+    end
```
Comment (tresorsawasawa) [inline spec/views/user_spec.rb]: - [ ] Good job implementing most of the tests required for this view. So, I noticed that you missed one of the [required tests](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md) to match the expected result. You should also add **test(s) that redirects to the post's show page when you click on a post.** Make sure you add it. But, Good job :+1:. 
Screenshot from 2022-11-16 15-39-52

File: spec/views/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      it 'displays how many likes a post has' do
+        visit user_posts_path
+        expect(page).to have_content('Likes: 0')
+      end
+
+      it 'When I click on a post, it redirects me to that post show page' do
+        visit user_posts_path
+        click_link 'slamander'
+        expect(page).to have_current_path("/users/#{@first_user.id}/posts/#{@first_post.id}/")
+      end
```
Comment (tresorsawasawa) [inline spec/views/post_spec.rb]: - [ ] As required for the project, kindly make sure you also add the below missed required tests for this view to match the expected result:
  - Tests that assert you can see the first comments on a post.
  - Tests that assert you can see a section for pagination if there are more posts than fit on the view.
Screenshot from 2022-11-16 15-48-06

File: spec/views/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+      it 'displays how many likes it has' do
+        visit user_post_path
+        expect(page).to have_content('Number of likes: 0')
+      end
+
+      it 'displays the post author' do
+        visit user_post_path
+        expect(page).to have_content(@first_user.Name)
+      end
```
Comment (tresorsawasawa) [inline spec/views/post_spec.rb]: - [ ] As required for the project, kindly make sure you also add the below missed required tests for this view to match the expected result and you can update the test to **display how many likes a post has** as it **is repeated twice**
  - Test(s) that asserts you can see the username of each **commentor**.
  - Test(s) that asserts you can see the comment each **commentor left**.
  
Screenshot from 2022-11-16 15-59-41

File: spec/examples.txt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+./spec/views/user_spec.rb[1:1:2]     | failed  | 2 minutes 13.7 seconds |
+./spec/views/user_spec.rb[1:1:3]     | unknown |                        |
+./spec/views/user_spec.rb[1:1:4]     | unknown |                        |
+./spec/views/user_spec.rb[1:2:1]     | unknown |                        |
+./spec/views/user_spec.rb[1:2:2]     | unknown |                        |
+./spec/views/user_spec.rb[1:2:3]     | unknown |                        |
+./spec/views/user_spec.rb[1:2:4]     | unknown |                        |
+./spec/views/user_spec.rb[1:2:5]     | unknown |                        |
+./spec/views/user_spec.rb[1:2:6]     | unknown |                        |
+./spec/views/user_spec.rb[1:2:7]     | unknown |                        |
```
Comment (tresorsawasawa) [inline spec/examples.txt]: - [ ] Kindly make sure all tests for your **views** are **passing**. When trying on my side no one of them has passed and if you can check for the result above no test for views has passed.
Below is the result I get from your views tests(i.e.: `rspec spec/views `)
Screenshot from 2022-11-16 16-11-13

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @PNdunguMaina 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**